File: src/greedy-algorithms/easy/minimum-waiting-time/go/sjf.go
package main

import (
	"sort"
)

func MinimumWaitingTime(queries []int) int {
	sort.Ints(queries)

	totalTime := 0
	for i, duration := range queries {
		remaining := len(queries) - (i + 1)
		totalTime += duration * remaining
	}

	return totalTime
}

File: src/greedy-algorithms/easy/minimum-waiting-time/js/sjf.js
// O(n.log(n)) time | O(1) space
function minimumWaitingTime(queries) {
    queries.sort((a, b) => a - b)

    let totalWaitingTime = 0
    for (let i = 0; i < queries.length; i++) {
        const duration = queries[i]
        const queriesLeft = queries.length - (i + 1)
        totalWaitingTime += duration * queriesLeft
    }

    return totalWaitingTime
}

exports.minimumWaitingTime = minimumWaitingTime;
